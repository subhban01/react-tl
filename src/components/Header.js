import React, {Component} from 'react';
import BEMHelper from 'react-bem-helper';
import moment from 'moment';

let header = new BEMHelper('header');
class Header extends Component {

    render() {
        const {companyInfo} = this.props;
        return (
            <section {...header()}>
                <div {...header('row1')}>
                    <span {...header('company-name')}>{companyInfo.companyName}</span>
                </div>
                <div {...header('row2')}>
                    <span {...header('company-motto')}>{companyInfo.companyMotto}</span>
                    <span {...header('company-estdate')}>Since {moment(companyInfo.companyEst).format('DD/MM/YYYY')}</span>
                </div>
            </section>
        );
    }
}

export default Header;