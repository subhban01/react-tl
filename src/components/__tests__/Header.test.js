import * as React from 'react';
import {shallow} from 'enzyme';
import Header from '../Header.js';


describe('Header Component', () => {
    it('should render the Header correctly', () => {
        const companyInfo = {
            "companyName": "Waters - Paterson",
            "companyMotto": "mission-critical whiteboard infomediaries",
            "companyEst": "2015-08-08T11:03:26.973Z"
        }

        const wrapper = shallow(
            <Header companyInfo={companyInfo}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
})
;