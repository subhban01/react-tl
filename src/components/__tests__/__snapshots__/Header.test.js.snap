// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render the Header correctly 1`] = `
<section
  className="header"
>
  <div
    className="header__row1"
  >
    <span
      className="header__company-name"
    >
      Waters - Paterson
    </span>
  </div>
  <div
    className="header__row2"
  >
    <span
      className="header__company-motto"
    >
      mission-critical whiteboard infomediaries
    </span>
    <span
      className="header__company-estdate"
    >
      Since 
      08/08/2015
    </span>
  </div>
</section>
`;
